libkbfs/tlf_handle.go

```diff
@@ -986,3 +523,51 @@ func (h TlfHandle) IsFinal() bool {
 func (h TlfHandle) IsConflict() bool {
 	return h.conflictInfo != nil
 }
+
+// GetPreferredFormat returns a TLF name formatted with the username given
+// as the parameter first.
+// This calls FavoriteNameToPreferredTLFNameFormatAs with the canonical
+// tlf name which will be reordered into the preferred format.
+// An empty username is allowed here and results in the canonical ordering.
+func (h TlfHandle) GetPreferredFormat(username libkb.NormalizedUsername) string {
+	s, err := FavoriteNameToPreferredTLFNameFormatAs(
+		username, string(h.GetCanonicalName()))
+	if err != nil {
+		panic("TlfHandle.GetPreferredFormat: Parsing canonical username failed!")
+	}
+	return s
+}
+
+// FavoriteNameToPreferredTLFNameFormatAs formats a favorite names for display with the
+// username given.
+// An empty username is allowed here and results in tlfname being returned unmodified.
+func FavoriteNameToPreferredTLFNameFormatAs(username libkb.NormalizedUsername,
+	tlfname string) (string, error) {
```

Comment:
Can `tlfname` be a `TlfCanonicalName` instead of a `string`?  Seems slightly safer.